rasa/graph_components/providers/domain_provider.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+from typing import Dict, Text, Any, Optional
+
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.core.domain import Domain
+from rasa.shared.importers.importer import TrainingDataImporter
+
+
+class DomainProvider(GraphComponent):
+    """Provides domain during training and inference time."""
+
+    def __init__(
+        self,
+        model_storage: ModelStorage,
+        resource: Resource,
+        domain: Optional[Domain] = None,
+    ) -> None:
+        """Creates domain provider."""
+        self._model_storage = model_storage
+        self._resource = resource
+        self._domain = domain
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> DomainProvider:
+        """Creates component (see parent class for full docstring)."""
+        return cls(model_storage, resource)
+
+    @classmethod
+    def load(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+        **kwargs: Any,
+    ) -> DomainProvider:
+        """Creates provider using a persisted version of itself."""
+        with model_storage.read_from(resource) as resource_directory:
+            domain = Domain.from_path(resource_directory)
+        return cls(model_storage, resource, domain)
+
+    def persist(self, domain: Domain) -> None:
```

Comment:
as it's not used outside of the class
```suggestion
    def _persist(self, domain: Domain) -> None:
```

Reply:
I also wanted to ask should I separate the tests to improve the coverage - atm that check fails because of coverage decrease?

Reply:
this test seems to be wrong - I just checked and you have a 100% test coverage. I fail to see how you could test better 😆 I'd just ignore the check for as it's not a required check.